frontend/server/src/main/java/com/amazonaws/ml/mms/http/HttpRequestHandler.java

```diff
@@ -61,6 +61,9 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest req) {
         } catch (ConflictStatusException e) {
             logger.trace("", e);
             NettyUtils.sendError(ctx, HttpResponseStatus.CONFLICT, e);
+        } catch (GatewayTimeoutException e) {
+            logger.trace("", e);
+            NettyUtils.sendError(ctx, HttpResponseStatus.GATEWAY_TIMEOUT, e);
```

Comment:
Probably 408 is a better. Thats a TIMEOUT_ERROR.

Reply:
Agreed, that fits better.